File: src/jvmgo/instructions/anewarray.go
package instructions

import "jvmgo/rtda"

// Create new array of reference
type anewarray struct {
    index uint16
}

func (self *anewarray) fetchOperands(bcr *BytecodeReader) {
    self.index = bcr.readUint16()
}

func (self *anewarray) execute(thread *rtda.Thread) {
    // todo
}

File: testclasses/src/main/java/instructions/ANewArray.java
package instructions;

public class ANewArray {
    
    public static void main(String[] args) {
        Object[] arr = new Object[8];
        if (arr.length == 8) {
            System.out.println("OK!");
        } else {
            System.out.println("Fail!");
        }
    }
    
}
